stdlib/3/tkinter/colorchooser.pyi

```diff
@@ -0,0 +1,7 @@
+from tkinter.commondialog import Dialog
+from typing import Any, ClassVar, Optional, Text, Tuple
+
+class Chooser(Dialog):
+    command: ClassVar[str]
+
+def askcolor(color: Optional[Text] = ..., **options: Any) -> Tuple[Optional[Tuple[float, float, float]], Optional[str]]: ...
```

Comment:
I think this function always returns either `(None, None)` or `((float, float, float), str)`

If you do this then it might help with checking which kind of return value it was: (if you apply the suggestion, then you also need to change the imports)

I'm not sure whether `color` should be `Text` or `str`, i.e. will this stub ever get ported to Python 2. Rest of tkinter stubs use `str`.

```suggestion
def askcolor(color: Optional[Text] = ..., **options: Any) -> Union[Tuple[None, None], Tuple[Tuple[float, float, float], str]]: ...
```

Reply:
The argument is `Text` not for py2 compat, but because the argument can be both str and bytes, from a quick test I did. As for the return.

Reply:
In Python 3 `Text` does *not* mean "str or bytes". It means just "str".

Reply:
Sorry, forgot that that special behavior is only valid in Python 2. (I misremembered from PEP 484, "For parameters typed as Text, arguments of type str as well as unicode should be acceptable." which is only applied when type-checking 2.7)

Reply:
Maybe many other things in tkinter should be changed from `str` to `Union[str, bytes]` too? I have used `str` for most things when working on the tkinter stubs.

Reply:
Probably, but I'd call that outside the scope of this PR. May look into doing it later.